assembl/static2/js/app/components/navbar/BurgerNavbar.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { browserHistory } from 'react-router';
+
+import Logo from './Logo';
+import NavigationMenu from './navigationMenu';
+import LanguageMenu from './languageMenu';
+import UserMenu from './UserMenu';
+
+export default class BurgerNavbar extends React.PureComponent {
+  componentWillMount() {
+    this.setState({ shouldDisplayMenu: false });
+    this.unlisten = browserHistory.listen(() => {
+      this.setState({ shouldDisplayMenu: false });
+    });
+  }
+  componentWillUnmount() {
+    this.unlisten();
+  }
+  toggleMenu = () => {
+    this.setState((prevState) => {
+      return {
+        shouldDisplayMenu: !prevState.shouldDisplayMenu
+      };
+    });
+  };
+  render() {
+    const { state, props, toggleMenu } = this;
+    const { elements, slug, logoSrc, connectedUserId, currentPhaseIdentifier, helpUrl, location } = props;
+    const { shouldDisplayMenu } = state;
+    return (
+      <div className="burger-navbar">
+        {shouldDisplayMenu && (
+          <div className="nav-burger-menu shown">
+            <NavigationMenu elements={elements} />
+            <LanguageMenu size="xl" className="burgermenu-language center" />
+          </div>
+        )}
+        <span onClick={toggleMenu} className={`burgermenu-icon assembl-icon-${shouldDisplayMenu ? 'cancel' : 'menu-on'} black`} />
```

Comment:
in this case, you could (should?) use `classNames()`

Reply:
Should, and yeah, this is perfect usecase for it.

Reply:
This way ?
```
const iconType = shouldDisplayMenu ? 'cancel' : 'menu-on';
const burgerButtonClassName = classNames({ [`assembl-icon-${iconType}`]: true }, ['burgermenu-icon', 'black']);
 ```
...
```
<span onClick={toggleMenu} className={burgerButtonClassName} />
```
Or this way ?
```
const burgerButtonClassNames = classNames(
      {
        'assembl-icon-menu-on': !shouldDisplayMenu,
        'assembl-icon-cancel': shouldDisplayMenu
      },
      ['burgermenu-icon', 'black']
    );
```
(...)
```
        <span onClick={toggleMenu} className={burgerButtonClassNames} />
```

Reply:
They both seems to be more convoluted ways to do the same thing.

Reply:
This way?
```
const burgerButtonClassNames = classNames([
      `assembl-icon-${shouldDisplayMenu ? 'cancel' : 'menu-on'}`,
      'burgermenu-icon',
      'black'
    ]);
```

Reply:
I don't understand why
```
className={classNames([`assembl-icon-${shouldDisplayMenu ? 'cancel' : 'menu-on'}`, 'burgermenu-icon', 'black'])
```
is better than
```
className={`burgermenu-icon assembl-icon-${shouldDisplayMenu ? 'cancel' : 'menu-on'} black`}
```

Reply:
lol yeah for this particuliar case where we have a dynamically generated class name, I don't see a benefit of using classNames.